spec.emu

```diff
@@ -337,29 +422,24 @@ contributors: Surma, Dan Ehrenberg
               1. Else,
                 1. Append _ee_ to _indirectExportEntries_.
             1. Let _async_ be _body_ Contains `await`.
-            1. <del>Return</del><ins>Let _moduleRecord_ be the</ins> Source Text Module Record { [[Realm]]: _realm_, [[Environment]]: ~empty~, [[Namespace]]: ~empty~, [[CycleRoot]]: ~empty~, [[HasTLA]]: _async_, [[AsyncEvaluation]]: *false*, [[TopLevelCapability]]: ~empty~, [[AsyncParentModules]]: &laquo; &raquo;, [[PendingAsyncDependencies]]: ~empty~, [[Status]]: ~unlinked~, [[EvaluationError]]: ~empty~, [[HostDefined]]: _hostDefined_, [[ECMAScriptCode]]: _body_, [[Context]]: ~empty~, [[ImportMeta]]: ~empty~, [[RequestedModules]]: _requestedModules_, [[ImportEntries]]: _importEntries_, [[LocalExportEntries]]: _localExportEntries_, [[IndirectExportEntries]]: _indirectExportEntries_, [[StarExportEntries]]: _starExportEntries_, [[DFSIndex]]: ~empty~, [[DFSAncestorIndex]]: ~empty~, <ins>[[TopLevelScriptOrModule]]: ~empty~</ins> }.
-            1. <ins>Set _moduleRecord_.[[TopLevelScriptOrModule]] to _moduleRecord_.</ins>
-            1. <ins>Return _moduleRecord_.</ins>
+            1. Return Source Text Module Record { [[Realm]]: _realm_, [[Environment]]: ~empty~, [[Namespace]]: ~empty~, [[CycleRoot]]: ~empty~, [[HasTLA]]: _async_, [[AsyncEvaluation]]: *false*, [[TopLevelCapability]]: ~empty~, [[AsyncParentModules]]: &laquo; &raquo;, [[PendingAsyncDependencies]]: ~empty~, [[Status]]: ~unlinked~, [[EvaluationError]]: ~empty~, [[HostDefined]]: _hostDefined_, <ins>[[HostDefinedResolutionBase]]: _hostDefinedResolutionBase_,</ins> [[ECMAScriptCode]]: _body_, [[Context]]: ~empty~, [[ImportMeta]]: ~empty~, [[RequestedModules]]: _requestedModules_, [[ImportEntries]]: _importEntries_, [[LocalExportEntries]]: _localExportEntries_, [[IndirectExportEntries]]: _indirectExportEntries_, [[StarExportEntries]]: _starExportEntries_, [[DFSIndex]]: ~empty~, [[DFSAncestorIndex]]: ~empty~ }.
           </emu-alg>
         </emu-clause>
 
         <emu-clause id="sec-gettoplevelscriptormodule" type="abstract operation">
           <h1>
             <ins>
-              GetTopLevelScriptOrModule (
+              GetHostDefinedResolutionBase (
```

Comment:
Interesting approach, could we make this entirely align with the `[[HostDefined]]` field already existing on module records?

Reply:
HTML's [[HostDefined]] keeps some data (for example, a list of unhandled rejected promises) that is not serializable, so copying it from the outer Module Record to ModuleBlock (so that then it can be copied to the ModuleBlock's Module Record) would make them non-serializable.

The idea here is to split the [[HostDefined]] part in two, so that we only carry over the (serializable) part needed to resolve modules but not the "runtime" data related to a specific Module Record.